src/test/java/org/simdjson/Utf8ValidatorTest.java

```diff
@@ -0,0 +1,498 @@
+package org.simdjson;
+
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.VectorSpecies;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.*;
+
+class Utf8ValidatorTest {
+    private static final VectorSpecies<Byte> VECTOR_SPECIES = ByteVector.SPECIES_256;
+
+
+    /* ASCII / 1 BYTE TESTS */
+
+    @Test
+    void validate_allEightBitValues_invalidAscii() {
+        byte[] invalidAscii = new byte[128];
+
+        int index = 0;
+        for (int eightBitVal = 255; eightBitVal >= 128; eightBitVal--) {
+            invalidAscii[index++] = (byte) eightBitVal;
+        }
+
+        SimdJsonParser parser = new SimdJsonParser();
+        for (int i = 0; i < 128; i += VECTOR_SPECIES.vectorByteSize()) {
```

Comment:
If I understand correctly, the purpose of this loop (in this and other tests) is to verify more invalid characters then, for example, the following code would do:

```java
        SimdJsonParser parser = new SimdJsonParser();
        assertThatExceptionOfType(JsonParsingException.class)
                .isThrownBy(() -> parser.parse(invalidAscii, invalidAscii.length))
                .withMessage("Invalid UTF8");
```
It would detect that the input is invalid after checking the first chunk.

I have three concerns:
* In my opinion, it's not immediately apparent why these loops exist and why they rely on the vector length.
* Although we test  more invalid characters, we don't cover all of them. If a chunk is 32 bits long and contains 32 invalid characters, effectively, only one character is tested. The rest of the characters might just as well be valid, and the validation result would be exactly the same.
* In most tests, we don't cover the part of `Utf8Validator::validate` that validates bytes that are not aligned with the vector width.

Please, let me know if you have an idea how to address the above points. I'll also think about it in the meantime.

Reply:
I'll address this in a separate PR.